ext/app/electron/config.ts

```diff
@@ -0,0 +1,206 @@
+  try {
+    let configBuffer = fse.readFileSync(filename)
+    config = new Config(ini.parse(configBuffer.toString("utf8")))
+  } catch (err) {
+    log.warn(`Failed to read configuration file: ${err}`)
+    config = new Config({})
+  }
+  // Section: login
+  config.apply(
+    "login.username",
+    "GRIST_DEFAULT_USERNAME",
+    NO_VALIDATION,
+    getUsername()
+  )
+  config.apply(
+    "login.email",
+    "GRIST_DEFAULT_EMAIL",
+    NO_VALIDATION,
+    getEmail()
+  )
+  // Section: server
+  config.apply(
+    "server.listen",
+    "GRIST_HOST",
+    NO_VALIDATION,
+    "localhost"
+  )
+  config.apply(
+    "server.port",
+    "GRIST_PORT",
+    (portstr) => {
+      if (! /^\d+$/.test(portstr)) {
+        return false
+      }
+      let port = parseInt(portstr)
+      return port > 0 && port < 65536
+    },
+    "_RANDOM"
+  )
+  config.apply(
+    "server.auth",
+    "GRIST_DESKTOP_AUTH",
+    (auth) => ["strict", "none", "mixed"].includes(auth),
+    "strict"
+  )
+  // Section: sandbox
+  config.apply(
+    "sandbox.flavor",
+    "GRIST_SANDBOX_FLAVOR",
+    (flavor) => ["pyodide", "unsandboxed", "gvisor", "macSandboxExec"].includes(flavor),
+    "pyodide"
+  )
+  // Section: storage
+  config.apply(
+    "storage.instance",
+    "GRIST_INST_DIR",
+    NO_VALIDATION,
+    APPDATA_DIR
+  )
+  config.apply(
+    "storage.documents",
+    "GRIST_DATA_DIR",
+    NO_VALIDATION,
+    electron.app.getPath("documents")
+  )
+  config.apply(
+    "storage.plugins",
+    "GRIST_USER_ROOT",
+    NO_VALIDATION,
+    path.join(electron.app.getPath("home"), ".grist")
+  )
+  config.apply(
+    "storage.homedb",
+    "TYPEORM_DATABASE",
+    NO_VALIDATION,
+    path.join(APPDATA_DIR, "home.sqlite3")
+  )
+
+  const homedb_location = path.parse((process.env.TYPEORM_DATABASE as string)).dir
+  if (!fse.existsSync(homedb_location)) {
+    log.warn(`Directory to contain the home DB does not exist, creating ${homedb_location}`)
+    fse.mkdirSync(homedb_location)
+  }
+
+  // We don't allow manually setting these envvars anymore. Fixing them makes maintaining grist-desktop easier.
+  process.env.GRIST_SINGLE_PORT = "true"
+  process.env.GRIST_SERVE_SAME_ORIGIN = "true"
+  process.env.GRIST_DEFAULT_PRODUCT = "Free"
+  process.env.GRIST_ORG_IN_PATH = "true";
+  process.env.GRIST_HIDE_UI_ELEMENTS = "helpCenter,billing,templates,multiSite,multiAccounts";
+  process.env.GRIST_CONTACT_SUPPORT_URL = packageJson.repository + "/issues"
+  if (process.env.GRIST_DESKTOP_AUTH !== "mixed") {
+    process.env.GRIST_FORCE_LOGIN = "true";
+  }
+
+  // Related to plugins (Would have to be changed if local custom widgets are used?)
+  suggestEnv("GRIST_WIDGET_LIST_URL", commonUrls.gristLabsWidgetRepository);
+}
+
+
+function getUsername(): string {
+  try {
+    return os.userInfo().username
+  } catch {
+    return "You"
+  }
+}
+
+function getEmail(): string {
+  return getUsername().toLowerCase() + "@" + os.hostname()
+}
+
+export {loadConfigFile}
```

Comment:
Can you just add `export` to `loadConfigFile` earlier.

Reply:
Still learning Typescript. It seems very versatile!